### ISSUE TITLE
[BUG] npm publish -ws is failing

### ISSUE THREAD

**Issue by manchuck** [npm/cli#7907]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

When you run `npm publish -ws`, it will go through and publish all those packages. However, if one of those packages is not updated, the command will fail since NPM rejects publishing the same version.

### Expected Behavior

Running `npm publish -ws` will keep publishing the other packages even if one fails

### Steps To Reproduce

Create a workspace with three packages (A, B, C)
Bump the versions for A and C 
Run `npm publish -ws`
B will fail to publish and npm will not try to publish C 

### Environment

- npm: 10.8.3
- Node.js: v22.9.0
- OS Name: MacOS
- System Model Name: Macbook Pro
- npm config:
```ini
; "user" config from /Users/manchuck/.npmrc

//registry.npmjs.org/:_authToken = (protected)
save-exact = true
save-prefix = ""

; node bin location = /Users/manchuck/.nvm/versions/node/v22.9.0/bin/node
; node version = v22.9.0
; npm local prefix = /Users/manchuck/Projects/vonage/node-sdk
; npm version = 10.8.3
; cwd = /Users/manchuck/Projects/vonage/node-sdk/packages
; HOME = /Users/manchuck
; Run `npm config ls -l` to show all defaults.

**Comment by manchuck** [comment]
I am opening this as an issue again because if this is intended, then it is poorly implemented. As a workaround for this, a user now has to run a bash script to publish packages when there is no change in the version number for one package in the monorepo:

```bash
for i in $(ls -d *)                                                                                                                          ─╯
(cd $i && npm publish)
```

I'm not sure how this is a breaking change for a fix. Even _if_ this was some breaking change, the remedy is to just add a new flag that can adjust the behavior without changing how it works.  Something like `--ignore-publish-fail`

**Comment by kchindam-infy** [comment]
@manchuck  Bumping the versions A and C and runs the npm publish -ws it publishes other packages even if one fails. It is working as you expected not getting any ERROR it publishes packages except which are not bumped or updated

**Comment by manchuck** [comment]
@kchindam-infy Correct, I am not getting an error when running the command; I am getting an error when trying to publish to npmjs.org since package B has not bumped its version.

**Comment by kchindam-infy** [comment]
@manchuck The publish will fail if the package name and version combination already exists in the specified registry Once a package is published with a given name and version, that specific name and version combination can never be used again, even if it is removed with [npm unpublis]. Refer to the documentation of npm publish which is given below 
https://docs.npmjs.com/cli/v11/commands/npm-publish.. its working by design .

**Comment by manchuck** [comment]
@kchindam-infy Yes, I know I cannot publish over an existing version. The problem is that I have failing actions because Package B did not need a version bump. A and C did. So, for me to publish A and C, I can't use `-ws`. Instead, I have to `(cd packages/A && npm publish)` and then `(cd packages/C && npm publish)`.

If we want to keep saying this is by design, no changes are needed. Then the `-ws` flag is useless. If that is so I will update my tooling to account for this

**Comment by kchindam-infy** [comment]
Closing the issue because it is working by the design
